src/Debug/WrappedDelayedListener.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Biig\Component\Domain\Debug;
+
+use Symfony\Component\VarDumper\Caster\ClassStub;
+
+class WrappedDelayedListener
+{
+    /**
+     * @var callable
+     */
+    private $listener;
+
+    /**
+     * @var string
+     */
+    private $pretty;
+
+    /**
+     * @var string Class function stub
+     */
+    private $stub;
+
+    /**
+     * @var mixed|string
+     */
+    private $name;
+
+    private static $hasClassStub;
+
+    /**
+     * WrappedDelayedListener constructor.
+     *
+     * @param $listener
+     */
+    public function __construct($listener)
+    {
+        $this->listener = $this->getListener($listener);
+
+        $this->name = \is_object($this->listener[0]) ? \get_class($this->listener[0]) : $this->listener[0];
+        $this->pretty = $this->name . '::' . $this->listener[1];
+
+        if (null === self::$hasClassStub) {
+            self::$hasClassStub = class_exists(ClassStub::class);
+        }
+    }
+
+    private function getListener($listener)
```

Comment:
Do `$listener` is always an object? You do a type check against the object type at line 40.

Otherwise, the following will fail

Reply:
yes it is, I add typehint